### PR TITLE
CAYCPLUGIN-39 Move Java logic to the front-end

### PR DESCRIPTION
**Pull request by andreyluiz**

### CODE REVIEW

File: src/main/ts/api.ts [hunk 1]
```diff
@@ -19,18 +19,58 @@
...
-  return getJSON(`${ROOT}/issues_creation_histogram`).then(({ history }: Response) =>
-    history.map(({ date, value }) => ({
-      x: parseISO(date),
-      y: value,
-    }))
-  );
+  const promises = [];
+  for (let i = MAX_YEARS - 1; i >= 0; i--) {
+    const date = getDateMinusYears(i);
+    promises.push(makeRequest(date));
```
Comment (andreyluiz) [inline src/main/ts/api.ts:56]: I just copied the behavior from the Java file, but I wonder if we could make this in a single request.

Comment (philippe-perrin-sonarsource) [reply src/main/ts/api.ts:55]: Let's keep it this way, merge for now, and discuss with Zipeng tomorrow. Indeed, it doesn't really make sense to do that, and the ticket description obviously does not mention anything about it...

**Review by philippe-perrin-sonarsource — APPROVED** [review]
LGTM. WFV.

You can rebase on the latest master to get the fix for the BBT QA.

**Review by philippe-perrin-sonarsource — CHANGES_REQUESTED** [review]
I actually forgot to mention that you now need to remove all non-necessary deps from the pom.xml file (which is why we are making this change).

**Review by philippe-perrin-sonarsource — APPROVED** [review]

**Pull request merged**